go-controller/pkg/cluster/gateway_localnet.go

```diff
@@ -190,11 +192,19 @@ func localnetIptRules(svc *kapi.Service) []iptRule {
 		}
 
 		nodePort := fmt.Sprintf("%d", svcPort.NodePort)
+		destination := ""
+		gwIPstr := strings.Split(localnetGatewayIP, "/")[0]
+		gwIP := net.ParseIP(gwIPstr)
+		if gwIP.To4() != nil {
+			destination = gwIPstr + ":" + nodePort
+		} else {
+			destination = "[" + gwIPstr + "]:" + nodePort
```

Comment:
you don't have to do all this. Just `net.JoinHostPort(destination, nodePort)`

Reply:
oh, nice.  makes sense that it's common enough problem to have a helper!  thanks

Reply:
fixed to use `net.JoinHostPort()` now